packages/react/src/alert/AlertIcon.js

```diff
@@ -24,9 +24,9 @@ const AlertIcon = forwardRef((inProps, ref) => {
   const styleProps = useAlertIconStyle({ variant, severity });
 
   const icon = useMemo(() => {
-    if (typeof iconProp === 'string') {
+    if (typeof iconProp === 'object') {
       return (
-        <Icon icon={iconProp} />
+        <Icon as={iconProp} />
```

Comment:
@Ginioo

The `icon` prop for the `Icon` component remains of type string in both v1 and v2. The default behavior will render the icon as-is if its type is neither `string` nor `undefined`. To pass a custom icon, use the following approach instead:

```jsx
<Alert variant="solid" severity="success" icon={<CheckCircleOIcon size="4x" />} >
  This is a success alert.
</Alert>
```